### DISCUSSION TITLE
self-host app access external css resource issue

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#4421] [Help] [answered]
Hello,

My Budibase was running on internal LAN without internet access, and some of my users also has restricted access to internet, then I found their access to my Budibase app was much slower than other users who has good internet. After doing a debug, I found that it's due to the access issue to fonts.googleapis.com, and I viewed page source code found below external css reference. My question is, is there any way to change those css file locally?

```
<link rel="stylesheet" href="https://rsms.me/inter/inter.css">
<link rel="preconnect" href="https://fonts.gstatic.com">
<link href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:wght@400;600;700&display=swap" rel="stylesheet">
<link href="https://cdn.jsdelivr.net/npm/remixicon@2.5.0/fonts/remixicon.css" rel="stylesheet">
```
The page loading takes about 20s, due to external css resource access issue
![image](https://user-images.githubusercontent.com/29649651/153547089-4b17d237-ffcf-4807-aeb9-162031bde7f7.png)

**Comment by mjashanks** [comment, marked as answer]
Unfortunately there is no way around this just now. We do have an issue logged for this - https://github.com/Budibase/budibase/issues/3164

I've linked this discussion in there.

**Reply by samg666** [reply]
@mjashanks Hi Michael, is it possible that I can find the released app files and replace those link with local or other faster mirrors?

**Reply by mjashanks** [reply]
I think the issue you would have is that the app files get rebuilt whenever you make a change to your app. However, I believe that they are stored in MinIO: https://docs.budibase.com/self-hosting/accessing-minio

**Reply by samg666** [reply]
thank you, I will try.
